bot/utils/messages.py

```diff
@@ -109,8 +110,19 @@ async def send_attachments(
         else:
             await destination.send(
                 embed=embed,
-                username=message.author.display_name,
+                username=sub_clyde(message.author.display_name),
                 avatar_url=message.author.avatar_url
             )
 
     return urls
+
+
+def sub_clyde(username: Optional[str]) -> Optional[str]:
```

Comment:
In what case `username` can be `None`? So much how I see, this should pass `str` every time.

Reply:
It's optional here though every use of that function passes a name.

https://github.com/python-discord/bot/blob/bcf6993de7de726683e6ca9b0f102b6ad1a732fa/bot/cogs/watchchannels/watchchannel.py#L199-L207

I don't see a problem with leaving it optional anyway.

Reply:
If this internally handles the `None` (and it does) then I agree the annotation is appropriate, but why does it handle it? Is it to be "shielded" from callers which may accidentally give a `None`? Won't that just push the error further, since something else that is expecting a string will get the declyded `None`?

Reply:
None is a valid value for webhook username (acts like an empty string and defaults to the webhook name) that may get passed in so I don't think there's any problem with letting None through.

Reply:
In that case I agree that this is perfectly fine.